vpr/src/pack/pack.cpp

```diff
@@ -378,13 +391,98 @@ static bool try_size_device_grid(const t_arch& arch, const std::map<t_logical_bl
         type_util[&type] = util;
 
         if (util > 1.) {
-            fits_on_device = false;
+            std::string block_name(type.name);
+            overused_block_types.push_back(block_name);
         }
         VTR_LOG("\tBlock Utilization: %.2f Type: %s\n", util, type.name);
     }
     VTR_LOG("\n");
+    return overused_block_types;
+}
+
+std::pair<enum e_unrel_clust_stat, enum e_unrel_clust_mode> parse_unrel_clust_from_str(std::string str) {
+    std::pair<enum e_unrel_clust_stat, enum e_unrel_clust_mode> conv_value;
+    if (str == "on") {
+        conv_value.first = e_unrel_clust_stat::ON;
+        conv_value.second = e_unrel_clust_mode::FIXED;
+    } else if (str == "off") {
+        conv_value.first = e_unrel_clust_stat::OFF;
+        conv_value.second = e_unrel_clust_mode::FIXED;
+    } else if (str == "auto") {
+        conv_value.first = e_unrel_clust_stat::OFF;
+        conv_value.second = e_unrel_clust_mode::AUTO;
+    } else {
+        std::stringstream msg;
+        msg << "Invalid conversion from '"
+            << str
+            << "' to e_unrel_clust_stat (expected one of: "
+            << "on, off, auto"
+            << ")";
+        VPR_FATAL_ERROR(VPR_ERROR_PACK, msg.str().c_str());
+    }
+    return conv_value;
+}
+static bool block_type_found(std::string block_type_name) {
```

Comment:
Add vertical space.

Ensure there is a comment on the earlier function declaration.